src/tiled/wangfiller.cpp

```diff
@@ -248,9 +250,9 @@ const Cell &WangFiller::getCell(const TileLayer &back,
                                 const QRegion &fillRegion,
                                 QPoint point) const
 {
-    if (!fillRegion.contains(point) && back.contains(point)) {
+    if (!fillRegion.contains(point) && (mIsInfinite || back.contains(point))) {
```

Comment:
Outside the fill region I guess we should always take the tile from `back`? Cause in that case we can drop the rest of the condition.